### ISSUE TITLE
Add Typescript type declarations

### ISSUE THREAD

**Issue by moshthepitt** [onaio/gisida#268]
We need to add Type declarations for everything that `gisida` exports, so that it can be used in a Typescript project sanely i.e. without making everything of type `any` in Typescript.

**Comment by cKellyDesign** [comment]
I think a good output of this ticket can be `@types/gisida` and `@types/gisida-react` which can be installed alongside gisida/-react in TypeScript projects.
```
$ yarn add gisida gisida-react @types/gisida-react @types/gisida-react
```

**Comment by moshthepitt** [comment]
I think the [definitelytyped](http://definitelytyped.org/) project is mostly for community contributed type definitions, but what we want is to eventually have Gisida type definitions distributed as part of Gisida itself.

This is the path that [Redux](https://github.com/reduxjs/redux) took: it started off as `@types/redux` but since Redux now ships with its own type definitions, installing `@types/redux` is not necessary and will actually get you a polite warning.

**Comment by cKellyDesign** [comment]
This is closed by #354
